sys/fmt/fmt.c

```diff
@@ -24,7 +24,8 @@
 #include <unistd.h>
 #include <string.h>
 
-#if defined(__WITH_AVRLIBC__) || defined(__mips__)
+#if defined(__WITH_AVRLIBC__) || defined(__mips__) || defined(MODULE_PICOLIBC)
```

Comment:
```suggestion
#if defined(__WITH_AVRLIBC__) || defined(__mips__) || defined(MODULE_PICOLIBC) || defined(MODULE_VFS)
```

?

Reply:
I'd rather fix the VFS integration than hack around it 😉

A simple offset to the VFS fds in the libc glue should do.
Will submit a patch soon. (The tricky part is testing it, seems like we have to test for read/write to fds together with VFS - that’ll be the largest part of the patch)